### PR TITLE
tc-build: Updates around Linux 5.6

### PR DESCRIPTION
**Pull request by nathanchance**
* Update the kernel build script to build Linux 5.6.

* Add support for s390 to all three scripts.

* Add support for RISC-V to `build-llvm.py` and `build-binutils.py`. RISC-V support will be added to the kernel build script when Linux 5.7 is out (see commit message for patches needed). Adding this now allows us to easily keep an eye on RISC-V in mainline and next.

* Add `mips-linux-gnu` to `build-binutils.py`; this is needed for MIPS big endian builds.

* Update the known good revision, as is tradition with Linux updates. There is a slight "regression" for certain Android devices (https://github.com/ClangBuiltLinux/linux/issues/958) but it is not reachable in normal scenarios because it requires forcibly enabling `CONFIG_LD_DEAD_CODE_DATA_ELIMINATION`.

This pull request has been run through [a test framework](https://github.com/nathanchance/tc-build/tree/82979e3c67b545fb5550f1a1248e060e975d47bd/test) that I am developing (and hope to eventually merge here), which builds a ton of kernels and boots them. I have experienced no visible regressions with this pull request.

### CODE REVIEW

File: build-binutils.py [hunk 1]
```diff
@@ -155,17 +158,20 @@ def invoke_configure(build_folder, install_folder, root_folder, target,
             'CXXFLAGS=-O2 -march=%s -mtune=%s' % (host_arch, host_arch)
         ]
     else:
-        configure += [
-            'CFLAGS=-O2',
-            'CXXFLAGS=-O2'
-        ]
+        configure += ['CFLAGS=-O2', 'CXXFLAGS=-O2']
```
Comment (msfjarvis) [inline build-binutils.py]: unrelated: Has anybody seen any benefit to experimenting with O3 here?

File: kernel/.gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 linux-*
 !*.patch
+!*.sha256
```
Comment (ojeda) [inline kernel/.gitignore]: Perhaps instead of adding negative rules, we could make the positive ones more restrictive, e.g. `kernel/linux-*`.

Comment (nathanchance) [reply kernel/.gitignore]: That will still cause `kernel/linux-*.tar.xz.sha256` to be ignored, which is what I am trying to avoid.

Comment (ojeda) [reply kernel/.gitignore]: Bad example then, I thought the other files were in the root folder. Then: `kernel/linux-*.tar.xz`.

Comment (nathanchance) [reply kernel/.gitignore]: The extracted folder (`linux-*`) needs to be ignored too, hence the initial exclusion. Realistically, this is the only other negative rule that we need for this scenario.

File: build-binutils.py [hunk 3]
```diff
@@ -77,7 +77,7 @@ def parse_parameters(root_folder):
                         help="""
                         The script can build binutils targeting arm-linux-gnueabi, aarch64-linux-gnu,
                         mipsel-linux-gnu, powerpc-linux-gnu, powerpc64-linux-gnu, powerpc64le-linux-gnu,
-                        and x86_64-linux-gnu.
+                        s390x-linux-gnu, and x86_64-linux-gnu.
```
Comment (kdrag0n) [inline build-binutils.py]: RISC-V is missing here.

Comment (nathanchance) [reply build-binutils.py]: Thanks, 5ddb83db1eeae4ca6e75db645afb288b20ea9bba

File: build-binutils.py (hunk 1 above)
Comment (Panchajanya1999) [reply build-binutils.py:161]: No, just build time is a bit longer.
I noticed this while building on Drone CI, over 48 cores. 
With O3, it exceeded time limit of 1 hr. While with O2, it built successfully in 57 minutes.

**Pull request merged**